test/integration/targets/iosxr_l2_interfaces/vars/main.yaml

```diff
@@ -0,0 +1,185 @@
+---
+merged:
+  before:
+    - name: loopback888
+    - name: loopback999
+    - name: GigabitEthernet0/0/0/1
+
+  commands:
+    - interface GigabitEthernet0/0/0/1
+    - dot1q native vlan 10
+    - l2transport l2protocol pvst tunnel
+    - l2transport l2protocol cdp forward
+    - l2transport propagate remote-status
+    - interface preconfigure GigabitEthernet0/0/0/1.2 l2transport
+    - dot1q vlan 10 any
+    - interface preconfigure GigabitEthernet0/0/0/3.900
+    - dot1q vlan 20 40
+
+  after:
+    - name: loopback888
+    - name: loopback999
+    - l2protocol:
+      - cdp: forward
+      - pvst: tunnel
+      l2transport: true
+      name: GigabitEthernet0/0/0/1
+      native_vlan: 10
+      propagate: true
+    - name: preconfigure GigabitEthernet0/0/0/1.2 l2transport
+      q_vlan:
+      - 10
+      - any
+    - name: preconfigure GigabitEthernet0/0/0/3.900
+      q_vlan:
+      - 20
+      - 40
+
+replaced:
+  before:
+    - name: loopback888
+    - name: loopback999
+    - l2protocol:
+      - cdp: forward
+      l2transport: true
+      name: GigabitEthernet0/0/0/1
+      native_vlan: 10
+      propagate: true
+    - name: preconfigure GigabitEthernet0/0/0/1.2 l2transport
+      q_vlan:
+      - 30
+      - any
+    - name: preconfigure GigabitEthernet0/0/0/3.900
+      q_vlan:
+      - 40
+      - 60
+    - l2protocol:
+      - vtp: tunnel
+      l2transport: true
+      name: GigabitEthernet0/0/0/4
+      native_vlan: 20
+
+  commands:
+    - interface GigabitEthernet0/0/0/1
+    - no l2transport
```

Comment:
Seems like we're setting `l2transport` later on for the same interface. This negate command is redundant?

Reply:
@NilashishC this is calling `no l2transport` as wrt to `l2transport` it appends to existing `l2transport` configuration and since for `replace` and `overridden` operation we change the existing config with provided config, it's needed to delete existing `l2transport` config and then apply the provided `l2transport` config.
ref: http://paste.openstack.org/show/761588/